Sort BranchVersions inside an Advisory

Jean85 opened the pull request:
This tries to address https://github.com/Roave/SecurityAdvisories/issues/40.

As a first step, I've added a single test that serves the branches in a non-sorted way and expects the same, sorted result of the test before.

Should I consider other use cases or add a `dataprovider` to the test to improve the robustness of this feature?

Ocramius commented on src/Roave/SecurityAdvisories/Advisory.php:
```diff
@@ -95,4 +95,24 @@ function (VersionConstraint $versionConstraint) {
+    private function sortVersionConstraints(array $versionConstraints): array
+    {
+        usort($versionConstraints, function(VersionConstraint $a, VersionConstraint $b) {
+            $versionA = $a->getLowerBound() ?? $a->getUpperBound();
```
The test doesn't cover constraints with ONLY lower bound or ONLY upper bound

Ocramius commented on src/Roave/SecurityAdvisories/VersionConstraint.php:
```diff
@@ -159,6 +159,18 @@ public function mergeWith(self $other) : self
         ));
     }
 
+    public static function sort(self $a, self $b): int
```
Is this method needed on this particular class? Looks like completely external?

Jean85 replied:
Nope, I put it there because it sorts only between elements of that class. If you prefer a separated class with `__invoke`, I'll do it! :+1:

Jean85 replied:
Done!

Jean85 commented:
I've added a dataprovider to test those cases, and found the underlying issue: `Component` needed testing, so I moved the sorting function as a static of `VersionConstraint` and use it in both `Advisory` and `Component`.

Now it should be good to go! In fact, this is the result testing locally (ommiting everything apart from affected lines):

```diff
{
    "conflict": {
-        "cakephp/cakephp": ">=2,<2.4.99|>=1.3,<1.3.18|>=3,<3.0.15|>=2.5,<2.5.99|>=2.6,<2.6.12|>=2.7,<2.7.6|>=3.1,<3.1.4"
+        "cakephp/cakephp": ">=1.3,<1.3.18|>=2,<2.4.99|>=2.5,<2.5.99|>=2.6,<2.6.12|>=2.7,<2.7.6|>=3,<3.0.15|>=3.1,<3.1.4",
-        "symfony/security": ">=2.3,<2.3.37|>=2.4,<2.6.13|>=2.7,<2.7.9|>=2,<2.0.25|>=2.1,<2.1.13|>=2.2,<2.2.9|>=2.7.30,<2.7.32|>=2.8.23,<2.8.25|>=3.2.10,<3.2.12|>=3.3.3,<3.3.5",
+        "symfony/security": ">=2,<2.0.25|>=2.1,<2.1.13|>=2.2,<2.2.9|>=2.3,<2.3.37|>=2.4,<2.6.13|>=2.7,<2.7.9|>=2.7.30,<2.7.32|>=2.8.23,<2.8.25|>=3.2.10,<3.2.12|>=3.3.3,<3.3.5",
    }
}
```

Ocramius commented:
@Jean85 seen the tests: pristine work, awesome! :+1:
